content/blog/how-to-locate-elements-in-playwright.md

```diff
@@ -0,0 +1,183 @@
+---
+title: How to locate elements in Playwright
+date: 2022-11-30
+description: How can we locate a link click on it in our e2e tests. In this article we will learn how to locate elements in Playwright using the test generator to generate your tets based on user actions or pick a locator by clicking on an element
+image: v1669817488/debbie.codes/blog/2022/vs-code-codegen_gxf58x.png
+alt: browser window showing highlighted element and inspector window with explore showing the locator highlighted
+ogImage: https://res.cloudinary.com/debsobrien/image/upload/v1669817488/debbie.codes/blog/2022/vs-code-codegen_gxf58x.png
+provider: cloudinary
+tags: [testing, playwright]
+published: true
+---
+
+In order to write end to end tests we need to first find elements on the webpage and then perform user actions on them. For example, find a link and click on it. 
+
+>The most convenient way to find an element is to use [Playwright's test generator](https://playwright.dev/docs/codegen-intro), which allows you to not only generate tests but also generate a locator for a specific element.
+
+But before we dive into how to use the test generator, first let's understand what a link element is and what exactly locators are.
+
+## What is a link?
+
+The HTML [`<a>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), also known as the anchor element, is an interactive element that creates a hyperlink to another page either within your site or externally to another site. It can also be use to link to specific areas on the same page as well as for emails, file downloads or anything else a URL can address. The `<a>` element uses the `href` attribute to pass in the URL that the hyperlink points to.
```

Comment:
```suggestion
The HTML [`<a>` element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a), also known as the anchor element, is an interactive element that creates a hyperlink to another page either within your site or externally to another site. It can also be used to link to specific areas on the same page as well as for emails, file downloads or anything else a URL can address. The `<a>` element uses the `href` attribute to pass in the URL that the hyperlink points to.
```